### ISSUE TITLE
CVE-2023-5363 for Debian trixie, but fix package version not available

### ISSUE THREAD

**Issue by isuftin** [anchore/grype#1604]
**What happened**:

We have been seeing CVE-2023-5363 pop up in our Debian Trixie Docker images.

```markdown
NAME     INSTALLED  FIXED-IN  TYPE  VULNERABILITY  SEVERITY 
libssl3  3.0.11-1   3.0.12-1  deb   CVE-2023-5363  High      
libssl3  3.0.11-1   3.0.12-2  deb   CVE-2023-5678  Unknown   
openssl  3.0.11-1   3.0.12-1  deb   CVE-2023-5363  High      
openssl  3.0.11-1   3.0.12-2  deb   CVE-2023-5678  Unknown
```

However, libssl3 and openssl 3.0.12 are not available for Debian trixie

**What you expected to happen**:

Grype does not show a fixed version for libssl3 or openssl until it exists

**How to reproduce it (as minimally and precisely as possible)**:

```Dockerfile
FROM debian:trixie-slim

RUN apt-get update && \
    apt-get install -y --no-install-recommends openssl
```

```sh
docker build -t my_test_image .

grype --only-fixed my_test_image:latest
 ✔ Vulnerability DB                [no update available]
 ✔ Loaded image                                                                                                                                                                                 my_test_image:latest
 ✔ Parsed image                                                                                                                              sha256:924d986e6f6e56663b554c582418f0f81865f48c91dc7b2dc21d25cc3bdf4314
 ✔ Cataloged packages              [89 packages]
 ✔ Scanned for vulnerabilities     [54 vulnerability matches]
   ├── by severity: 1 critical, 5 high, 0 medium, 1 low, 47 negligible
   └── by status:   4 fixed, 50 not-fixed, 0 ignored
NAME     INSTALLED  FIXED-IN  TYPE  VULNERABILITY  SEVERITY
libssl3  3.0.11-1   3.0.12-2  deb   CVE-2023-5678  High
libssl3  3.0.11-1   3.0.12-1  deb   CVE-2023-5363  High
openssl  3.0.11-1   3.0.12-2  deb   CVE-2023-5678  High
openssl  3.0.11-1   3.0.12-1  deb   CVE-2023-5363  High
```

**Anything else we need to know?**: https://packages.debian.org/search?keywords=openssl

**Environment**:
- Output of `grype version`: grype 0.73.1
- OS (e.g: `cat /etc/os-release` or similar): MacOS 13.6.1, Also seeing this in Grype installed in Alpine using a nightly db

**Comment by willmurphyscode** [comment]
Hi @isuftin, thanks very much for the report! Let me share a little background on why this started happening.

When Grype (via Syft) scans an image, it determines the namespace of operating system package vulnerabilities by looking at the `/etc/os-release` file. We recently fixed #1446 so that vulnerabilities in testing or unstable Debian could be detected at all. 

Now, we map `/etc/os-release` files that look like this:

``` ini
PRETTY_NAME="Debian GNU/Linux trixie/sid"
NAME="Debian GNU/Linux"
VERSION_CODENAME=trixie
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```

To our vulnerability namespace for Debian unstable. 

Unfortunately, testing and unstable have exactly the same `/etc/os-release` file, so we're not able to detect which one we're scanning. This means that if a fix is available for unstable, but not for testing, Grype will report the vulnerability as having a fix available:

``` sh
$ diff <(docker run --rm debian:unstable-slim cat /etc/os-release) \
   <(docker run --rm debian:trixie-slim cat /etc/os-release)
# no diff
```

I think this is probably a limitation in Debian's labelign system - it would be nice if we could tell testing and unstable apart when scanning an image, but given that we can't right now, I'm not sure it makes sense to disregard info on fixes that are available in unstable.

Does that make sense? Once trixie is released, Debian will start putting a version ID in the `/etc/os-release` file for trixie, and then we'll be able to distinguish trixies from unstable.

**Comment by isuftin** [comment]
@willmurphyscode That does make sense, thanks. That is quite unfortunate. On our end we'll go ahead and add the vulnerability to an ignore list for Trixie and monitor when the fixed package does hit the repos for Trixie. Thanks for the explanation!
